packages/data-service/src/__tests__/graphqlObservable-test.js

```diff
@@ -0,0 +1,218 @@
+      const result = graphqlObservable(query, schema, {
+        query: dataSource,
+        nameFilter
+      });
+
+      m.expect(result.take(1)).toBeObservable(expected);
+    });
+
+    itMarbles("filters by static argument", function(m) {
+      const query = gql`
+        query {
+          launched(name: "discovery") {
+            name
+            firstFlight
+          }
+        }
+      `;
+
+      const expectedData = [{ name: "discovery" }, { name: "challenger" }];
+      const dataSource = Observable.of(expectedData);
+      const expected = m.cold("(a|)", { a: { launched: [expectedData[0]] } });
+
+      const result = graphqlObservable(query, schema, {
+        query: dataSource
+      });
+
+      m.expect(result.take(1)).toBeObservable(expected);
+    });
+
+    itMarbles("filters out fields", function(m) {
+      const query = gql`
+        query {
+          launched {
+            name
+          }
+        }
+      `;
+
+      const expectedData = [{ name: "discovery", firstFlight: 1984 }];
+      const dataSource = Observable.of(expectedData);
+      const expected = m.cold("(a|)", {
+        a: { launched: [{ name: "discovery" }] }
+      });
+
+      const result = graphqlObservable(query, schema, {
+        query: dataSource
+      });
+
+      m.expect(result.take(1)).toBeObservable(expected);
+    });
+
+    itMarbles("filters out fields", function(m) {
+      const query = gql`
+        query {
+          launched {
+            name
+          }
+        }
+      `;
+
+      const expectedData = [{ name: "discovery", firstFlight: 1984 }];
+      const dataSource = Observable.of(expectedData);
+      const expected = m.cold("(a|)", {
+        a: { launched: [{ name: "discovery" }] }
+      });
+
+      const result = graphqlObservable(query, schema, {
+        query: dataSource
+      });
+
+      m.expect(result.take(1)).toBeObservable(expected);
+    });
+
+    itMarbles("resolve with query alias", function(m) {
+      const query = gql`
+        query nasa {
+          launched {
+            name
+          }
+        }
+      `;
+
+      const expectedData = [{ name: "discovery", firstFlight: 1984 }];
+      const dataSource = Observable.of(expectedData);
+      const expected = m.cold("(a|)", {
+        a: { nasa: [{ name: "discovery" }] }
+      });
+
+      const result = graphqlObservable(query, schema, {
+        query: dataSource
+      });
+
+      m.expect(result.take(1)).toBeObservable(expected);
+    });
+
+    itMarbles("resolve multiple queries with alias", function(m) {
+      const query = gql`
+        query nasa {
+          launched {
+            name
+          }
+        }
+
+        query roscosmos {
+          launched {
+            name
+          }
+        }
+      `;
+
+      const expectedData = [{ name: "discovery", firstFlight: 1984 }];
+      const dataSource = Observable.of(expectedData);
+      const expected = m.cold("(a|)", {
```

Comment:
Would it be hard to add a test for changing data and not "only" one-time data being exposed?

Reply:
Could you do it maybe in a separate PR? Would love to see it 👍

Reply:
Yes. I will create a ticket to do that. I think it is very valuable. There is an approach to copy the  tests from `graphql-js` so we can be sure we are as good as they are.